matching-tools/src/main/java/com/graphhopper/matching/tools/Measurement.java

```diff
@@ -0,0 +1,228 @@
+        MiniPerfTest miniPerf = new MiniPerfTest() {
+            @Override
+            public int doCalc(boolean warmup, int run) {
+                boolean foundPath = false;
+
+                // keep going until we find a path (which we may not for certain start/end points)
+                while (!foundPath) {
+
+                    // create random points and find route between:
+                    double lat0 = bbox.minLat + rand.nextDouble() * latDelta;
+                    double lon0 = bbox.minLon + rand.nextDouble() * lonDelta;
+                    double lat1 = bbox.minLat + rand.nextDouble() * latDelta;
+                    double lon1 = bbox.minLon + rand.nextDouble() * lonDelta;
+                    GHResponse r = hopper.route(new GHRequest(lat0, lon0, lat1, lon1));
+
+                    // if found, use it for map mathching:
+                    if (!r.hasErrors()) {
+                        foundPath = true;
+                        long time = 0;
+                        double sampleProportion = rand.nextDouble();
+                        GHPoint prev = null;
+                        List<GPXEntry> mock = new ArrayList<GPXEntry>();
+                        PointList points = r.getBest().getPoints();
+                        // loop through points and add (approximately) sampleProportion of them:
+                        for (GHPoint p : points) {
+                            if (null != prev && rand.nextDouble() < sampleProportion) {
+                                // estimate a reasonable time taken since the last point, so we
+                                // can give the GPXEntry a time. Use the distance between the
+                                // points and a random speed to estimate a time.
+                                double dx = distCalc.calcDist(prev.lat, prev.lon, p.lat, p.lon);
+                                double speedKPH = rand.nextDouble() * 100;
+                                double dt = (dx / 1000) / speedKPH * 3600000;
+                                time += (long) dt;
+                                // randomise the point lat/lon (i.e. so it's not
+                                // exactly on the route):
+                                GHPoint randomised = distCalc.projectCoordinate(p.lat, p.lat,
+                                        20 * rand.nextDouble(), 360 * rand.nextDouble());
+                                mock.add(new GPXEntry(randomised, time));
+                            }
+                        }
+                        // now match, provided there are enough points
+                        if (mock.size() > 2) {
+                            mapMatching.doWork(mock);
+                        } else {
+                            foundPath = false; // retry
+                        }
+
+                        // TODO: do we need to return something non-trivial?
+                        return 0;
```

Comment:
for all return values we need some results from the calculation to avoid JVM optimization kick in and remove the whole code above (although unlikely here IMO)